sqlmesh/dbt/common.py

```diff
@@ -156,15 +176,15 @@ class Dependencies(PydanticModel):
         refs: The table_name for models used
     """
 
-    macros: t.Set[MacroReference] = set()
-    sources: t.Set[str] = set()
-    refs: t.Set[str] = set()
+    macros: t.List[MacroReference] = []
+    sources: t.List[str] = []
+    refs: t.List[str] = []
 
     def union(self, other: Dependencies) -> Dependencies:
         return Dependencies(
-            macros=self.macros | other.macros,
-            sources=self.sources | other.sources,
-            refs=self.refs | other.refs,
+            macros=list(set(self.macros) | set(other.macros)),
```

Comment:
should we use the unique helper so we don't lose order

Reply:
It was all sets before, so I don't think this matters.

Reply:
Is this a serialization workaround?